Introduce config parameter for default DisplayNameGenerator (#1823)
junit-team/junit5: documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

      '-- Year 2048 is a leap year. [OK]
```

> [[override-default-display-name-generator]]

Reviewer: We never use generic anchors like that. Instead, all of our anchors are hierarchical according to the topic.

For example, `writing-tests-display-name-generator-default` would be more appropriate

Author: Thank you for the information. I will change accordingly.

==== Global configuration to override the default DisplayNameGenerator
Along with using `@DisplayName` and `@DisplayNameGeneration` annotation for custom display name
, it is possible to change the default display name generator for
the execution of an entire test plan. To change the default display name generator,
set the `junit.jupiter.displayname.generator.default` configuration parameter to the fully qualified
name of an class implementing `org.junit.jupiter.api.DisplayNameGenerator`.
The configuration can be supplied as a JVM system property, as a _configuration parameter_ in the
`LauncherDiscoveryRequest` that is passed to the `Launcher`, or via the JUnit Platform

> configuration file (see <<running-tests-config-params>> for details).

Reviewer: Please wrap at 90 characters and add blank lines between paragraphs.

Author: oops :)

For example, to use `DisplayNameGenerator.ReplaceUnderscores` as a default display name generator
you can start your JVM with the following system property.

> `-Djunit.jupiter.displayname.generator.default=org.junit.jupiter.api.DisplayNameGenerator.DisplayNameGenerator.ReplaceUnderscores`

Reviewer: That class name is incorrect: one too many `DisplayNameGenerator` elements.

Author: Oops

> make sure you use fully qualifies name (with class with package name) for the class and the class should

Reviewer: ```suggestion
Make sure you supply a fully qualified class name and that the class
```

> implement `org.junit.jupiter.api.DisplayNameGenerator` interface.

Reviewer: ```suggestion
implements the `org.junit.jupiter.api.DisplayNameGenerator` interface.
```

precedence of display name will be
1. `@DisplayName` annotation
2. Global jupiter configuration parameter `junit.jupiter.displayname.generator.default`

> 3. `@DisplayNameGeneration` annotation

Reviewer 2: :x: 2 and 3 should be the other way around.

Author: acknowledged
